Core/include/Acts/Geometry/ConeVolumeBounds.hpp

```diff
@@ -0,0 +1,228 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2020 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+#include "Acts/Geometry/Volume.hpp"
+#include "Acts/Geometry/VolumeBounds.hpp"
+#include "Acts/Utilities/BoundingBox.hpp"
+#include "Acts/Utilities/Definitions.hpp"
+
+#include <cmath>
+
+namespace Acts {
+
+class Surface;
+class CylinderBounds;
+class ConeBounds;
+class RadialBounds;
+class PlanarBounds;
+class Volume;
+
+/// @class ConeVolumeBounds
+///
+/// Volume bound class for describing conical volumes
+/// either with cylindrical inlay or outer boundary, it also allows
+/// for a sectoral description
+class ConeVolumeBounds : public VolumeBounds {
+ public:
+  /// @enum BoundValues for readability
+  enum BoundValues : unsigned int {
+    eInnerAlpha = 0,
+    eInnerOffsetZ = 1,
+    eOuterAlpha = 2,
+    eOuterOffsetZ = 3,
+    eHalfLengthZ = 4,
+    eAveragePhi = 5,
+    eHalfPhiSector = 6,
+    eSize = 7
+  };
+
+  ConeVolumeBounds() = delete;
+
+  /// Constructor - for general cone-cone setups
+  ///
+  /// @param innerAlpha The opening angle of the inner cone (0 if no cone)
+  /// @param innerOffsetZ The tip  z position in of the inner cone, w.r.t center
+  /// @param outerAlpha  The opening angle of the outer cone (0 if no cone)
+  /// @param outerOffsetZ The tip  z position in of the outer cone, w.r.t center
+  /// @param halflengthZ The minimum z value of the inner and outer cones
+  /// @param avergePhi The phi orientation of the sector
+  /// @param halfPhiSector The opening angle phi sector
+  ConeVolumeBounds(double innerAlpha, double innerTipZ, double outerAlpha,
+                   double outerOffsetZ, double halflengthZ, double averagePhi,
+                   double halfPhiSector) noexcept(false);
```

Comment:
What is motivation for adding `noexcept(false)` here? If I read the [noexcept spec](https://en.cppreference.com/w/cpp/language/noexcept_spec) correctly, this means that the constructor is not exception safe. That is the default anyways so `noexcept(false)` has no effect at all.

Reply:
I *think* clang requests that explicitly, that's why I added, I can test - but if we don't need it, I'd prefer a separate PR for removing it consistently in Geometry and Surfaces module.

Reply:
Will check - and if not needed I will make a separate PR for Geometry/Surfaces.

Reply:
Will be followed up.